xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/LocalMUCUser.java

```diff
@@ -574,6 +575,76 @@ private void processDecliningInvitationMessage(
                                          info.elementTextTrim("reason"), packet.getFrom());
     }
 
+    /**
+     * This method will parse a packet and checks if it is a MEMBER! list retrieval packet or not. 
+     * @param packet IQ
+     * @return true, if it is or false if not
+     */
+    private boolean isListRetrieval(IQ packet)
```

Comment:
There might be a security problem here, as, when this method returns 'true', the packet is processed as if it is sent by an admin. That's dangerous. Can we somehow do that differently?

Also, this implementation does not check a couple of things:
- is this a retrieval _request_ (is it of type 'get')? This method will return 'true' even for a 'set' - would that allow people to apply changes, even if they're not supposed to?
- are there _other_ directives (apart from requesting the member list) in this stanza? Can this code be abused to ask for the member list _as well as something else_ to bypass this check?

Reply:
I am now checking for a request packet...
Furthermore the packet will be checked for not allowed tags... otherwise an exception will be thrown